### DISCUSSION TITLE
What's missing in Ender3S1-F4-UBL-CVLM-20231215.bin?

### DISCUSSION THREAD

**Discussion by TheWebMachine** [mriscoc/Ender3V2S1#1199] [Q&A]
So, I suppose you are still working out the ABL2.0 F4 issues, but I was surprised to see an F4 UBL CVLM version. I thought we couldn't fit UBL with CVLM in the F4's limited space. What all additional did you have to remove beyond what had to be removed in 202309 CVLM ABL version? Beyond ABL/UBL, what are the major differences? Just want to make sure I don't try to employ a feature that might be missing.

**Comment by mriscoc** [comment]
Yes, but I cannot identify why non UBL F4 firmwares have boot problems, I know that it is a problem with the CRC possibly caused by the use of the numeric coprocessor of the F4. There is an open issue (https://github.com/mriscoc/Ender3V2S1/issues/1180) where some people have successfully compiling a ABL version of F4 by removing the code for the TMC driver parameters, but that parameters are not used in the F4 board.

**Comment by mriscoc** [comment]
These are the configuration files used for the Ender3S1-F4-UBL-CVLM: 
[Ender3S1-F4-UBL-CVLM.zip](https://github.com/mriscoc/Ender3V2S1/files/13824380/Ender3S1-F4-UBL-CVLM.zip)
